src/transformers/integrations/awq.py

```diff
@@ -241,7 +243,27 @@ def fuse_awq_modules(model, quantization_config):
         _fuse_awq_mlp(model, name, modules_to_fuse["mlp"], module, QuantFusedMLP)
 
         # Replace attention layers
-        _fuse_awq_attention_layers(model, module, modules_to_fuse, name, QuantAttentionFused)
+        attention_has_been_fused = _fuse_awq_attention_layers(
+            model, module, modules_to_fuse, name, QuantAttentionFused
+        )
+
+        if attention_has_been_fused:
+            fused_attention_modules.append(name)
+
+    # For AWQ fused + Llama we need to set `config._attn_implementation` = "custom" to avoid unexpected behavior and pass
+    # `None` attention mask to the fused attention modules as now the attention mask is dropped by our models and dealt
+    # by the `AttentionMaskConverter` module.
+    if len(fused_attention_modules) > 0:
+        fused_attention_parent_modules = {
+            fused_attention_module.split(".")[0] for fused_attention_module in fused_attention_modules
+        }
+        for module_name, module in model.named_modules():
+            if any(
+                module_name in fused_attention_parent_module
+                for fused_attention_parent_module in fused_attention_parent_modules
+            ):
+                if hasattr(module, "config") and hasattr(module.config, "_attn_implementation"):
+                    module.config._attn_implementation = "custom"
```

Comment:
because some models such as llava have other components such as the CLIP vision backbone, we want to keep them untouched and use the passed attn_implementation